Polyhedron/demo/Polyhedron/Plugins/Surface_mesh_deformation/Edit_polyhedron_plugin.cpp

```diff
@@ -457,6 +459,13 @@ Polyhedron_demo_edit_polyhedron_plugin::convert_to_plain_polyhedron(Item_id i,
   Scene_polyhedron_item* poly_item = edit_item->to_polyhedron_item();
   scene->replaceItem(i, poly_item);
   delete edit_item;
+  if(saved_color.size() >0)
```

Comment:
You did the same mistake I did. color_vector() returns the vector Colors_
which is not the vector that is cleared with invalidateOpenGLBuffers but
the color_map that gives the color of the facets according to its patch_id.

2016-06-27 15:31 GMT+02:00 Sebastien Loriot <notifications@github.com>:

> In
> Polyhedron/demo/Polyhedron/Plugins/Surface_mesh_deformation/Edit_polyhedron_plugin.cpp
> <https://github.com/CGAL/cgal/pull/1209#discussion_r68575992>:
>
> > @@ -457,6 +459,13 @@ Polyhedron_demo_edit_polyhedron_plugin::convert_to_plain_polyhedron(Item_id i,
> >    Scene_polyhedron_item* poly_item = edit_item->to_polyhedron_item();
> >    scene->replaceItem(i, poly_item);
> >    delete edit_item;
> > +  if(saved_color.size() >0)
>
> Quickly reading the code of the draw, it seems useless to save the color
> vector since calling invalidateOpenGLBuffers() will trigger a call to
> compute_normals_and_vertices() that will fill (overwrite) that vector.
> Simply drop the local color vector and the patch should be sufficient.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/1209/files/b265d26a53ba916ef04c2c0d99ed26091f98bb70#r68575992>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKyW1ZOVZF8gsMUD8XrT2UmU5ag8RNIAks5qP9C7gaJpZM4I_FmU>
> .
>